pylint/checkers/variables.py

```diff
@@ -1890,29 +1889,36 @@ def _loopvar_name(self, node: astroid.Name) -> None:
         # the variable is not defined.
         scope = node.scope()
         if isinstance(scope, nodes.FunctionDef) and any(
-            asmt.statement().parent_of(scope) for asmt in astmts
+            asmt.scope().parent_of(scope) for asmt in astmts
         ):
             return
-
-        # filter variables according their respective scope test is_statement
-        # and parent to avoid #74747. This is not a total fix, which would
+        # Filter variables according to their respective scope. Test parent
+        # and statement to avoid #74747. This is not a total fix, which would
         # introduce a mechanism similar to special attribute lookup in
         # modules. Also, in order to get correct inference in this case, the
         # scope lookup rules would need to be changed to return the initial
         # assignment (which does not exist in code per se) as well as any later
         # modifications.
+        # pylint: disable-next=too-many-boolean-expressions
```

Comment:
```suggestion
        # pylint: disable-next=too-many-boolean-expressions
        parent_is_root_module = astmts[0].parent == astmts[0].root()
```
```suggestion
        # pylint: disable-next=too-many-boolean-expressions
        #  If parent is the root Module and the parent of node we don't do 
        # _astmts = astmts[:1].
```

Reply:
First suggestion: Perhaps Github is not working correctly but we can't access `astmts[0]` on L1903. We need the `not astmts` check on L1904 before it.

Second suggestion: Is this necessary? That's basically explaining what the if statement is telling you.

Reply:
I suggested that because you precised this for the review so we might as well put a permanent comment ?

Reply:
Hm yeah, I don't actually specifically known why we don't want to do it.. There is a reference to an issue number of an old numbering system which I can't seem to find and I didn't immediately understand the test that was added in the same commit.

Unless somebody comes in and comes up with an actual explanation for why we don't want to do `_astmts = astmts[:1]` in this case I think we should leave this as is. I added the comment to make reviewing this easier.

Reply:
Make sense, I'm resolving :)